packages/ui/src/components/editor/input/Euler/index.tsx

```diff
@@ -51,13 +51,20 @@ const getBoundedRoundedAngle = (angle: number) => {
 }
 
 const tempEuler = new Euler() // we need the persistance, the hookstate doesnt register the dynamically allocated euler and quat value otherwise, thus we cannot assign new variable to the same
+const tempQuat = new Quaternion()
 export const EulerInput = (props: EulerInputProps) => {
+  tempQuat.copy(props.quaternion)
   tempEuler.setFromQuaternion(props.quaternion, 'YXZ')
   const angle = useHookstate({
     x: getBoundedRoundedAngle(tempEuler.x * RAD2DEG),
     y: getBoundedRoundedAngle(tempEuler.y * RAD2DEG),
     z: getBoundedRoundedAngle(tempEuler.z * RAD2DEG)
   })
+  angle.set({
```

Comment:
it doesn't make sense to have a hookstate that is immediately set. we can avoid it entirely and use the values directly

Reply:
sure, ill test if that works

Reply:
it works , also fixed an issue, material entities do not have a transform component associated with them but do have an entitytree, therefore errors popup when we compute transforms for the material entities parent but fail for the material entity itsef as it doesn not have a tranform component, so i made sure transforms are only computed for children entities which have a transform component